Fix #1034: workflow/bot operations API

harshach opened the pull request:
Ready to review

mithmatt commented on catalog-rest-service/src/main/resources/json/operations/workflows/ingestion.json:
```diff
@@ -0,0 +1,171 @@
+      "default": 300
+    },
+    "workflowCatchup": {
+      "description": "Workflow catchup for past executions.",
```
What does "catchup" refer to for ingestion?

harshach replied:
catchup as in , run for past days if the start date is scheduled for few days back
this will be useful in usage related workflows where we need to analyze the queries.

mithmatt replied:
Thanks for describing this. 

I believe another common term for this is backfill.

May be a better description (since this goes into documentation):
"Run past executions if the start date is in the past" (describes the term catchup or backfill, without using the term themselves)

mithmatt commented on catalog-rest-service/src/main/resources/json/schema/operations/workflows/createWorkflow.json:
```diff
@@ -0,0 +1,175 @@
+      "default": null
+    },
+    "owner": {
+      "description": "Owner of this database",
```
workflow instead of database

mithmatt commented on catalog-rest-service/src/main/resources/json/schema/operations/workflows/createWorkflow.json:
```diff
@@ -0,0 +1,175 @@
+    },
+    "owner": {
+      "description": "Owner of this database",
+      "type": "string"
```
Should this be: `      "$ref": "../../type/entityReference.json",` instead of string

harshach replied:
I am going to remove the createWorkflow here. There needs to be refactoring a bit here.
However in here we are capturing the connection details to connect to a data source such as hive, redshift etc.. to fetch the metadata about databases, tables . so having this as reference to an entity which is not created yet would be confusing.

mithmatt commented on catalog-rest-service/src/main/resources/json/schema/operations/workflows/ingestion.json:
```diff
@@ -0,0 +1,238 @@
+        },
+        {
+          "name": "mssql"
+        }, {
```
Indentation of { is off

harshach commented:
@mithmatt addressed your comments.
@sureshms please review as well. There are some shortcuts which we need to clean up based on how the service and connection will evolve in next release.
